napalm/ios/ios.py

```diff
@@ -553,8 +596,61 @@ def commit_config(self, message="", revert_in=None):
         # After a commit - we no longer know whether this is configured or not.
         self.prompt_quiet_configured = None
 
-        # Save config to startup (both replace and merge)
-        output += self.device.save_config()
+        if revert_in is None:
+            # Save config to startup (both replace and merge)
+            output += self.device.save_config()
+
+    def _check_archive_feature(self):
+        cmd = "show archive"
+        output = self.device.send_command_expect(cmd)
+        if "Archive feature not enabled" in output:
+            return False
+        return True
+
+    def has_pending_commit(self):
+        pending_commits = self._get_pending_commits()
+        return bool(pending_commits)
+
+    def _get_pending_commits(self):
+        if self._check_archive_feature():
+            cmd = "show archive config rollback timer"
+            output = self.device.send_command_expect(cmd)
```

Comment:
Also _get_pending_commits() should return `timer` in seconds (as opposed to a string). This will make it more consistent with the behavior of the `junos` and `eos` driver.

Reply:
Never mind...we can just leave the timer as-is. That should be fine.